### ISSUE TITLE
OracleJRE Upgrade

### ISSUE THREAD

**Issue by lawrencegripper** [opencb/opencga#1095]
Currently the docker images are build with OpenJDK 8 and run with 8. 

We think this causes issue #1076

To resolve we're looking at updating the JDK versions.

**Comment by lawrencegripper** [comment]
Constraints:

- We can only build successfully against 8
- Versions 9 and 10 aren't in mainstream repositories or dockerhub images so are untested as yet for build or run. 
- We know we can't build with 8 and run against >10 because of the `JJWT` dependency on Javax which was removed in Java 11
- We can't build with 11 because we see implicit typing errors
-

**Comment by jjcollinge** [comment]
Building OpenJDK8 and running on OpenJDK9 with additional poms.xml in opencga-catalog:
```shell
<dependency>
    <groupId>javax.xml.bind</groupId>
    <artifactId>jaxb-api</artifactId>
    <version>2.2.11</version>
</dependency>
<dependency>
    <groupId>com.sun.xml.bind</groupId>
    <artifactId>jaxb-core</artifactId>
    <version>2.2.11</version>
</dependency>
<dependency>
    <groupId>com.sun.xml.bind</groupId>
    <artifactId>jaxb-impl</artifactId>
    <version>2.2.11</version>
</dependency>
<dependency>
    <groupId>javax.activation</groupId>
    <artifactId>activation</artifactId>
    <version>1.1.1</version>
</dependency>
<dependency>
    <groupId>javax.annotations</groupId>
    <artifactId>javax.annotation-api</artifactId>
    <version>1.3.2</version>
</dependency>
```
results in:
```shell
exception in thread "main" java.lang.NoClassDefFoundError: javax/annotation/Priority
        at org.glassfish.jersey.model.internal.ComponentBag.modelFor(ComponentBag.java:483)
        at org.glassfish.jersey.model.internal.ComponentBag.access$100(ComponentBag.java:89)
        at org.glassfish.jersey.model.internal.ComponentBag$5.call(ComponentBag.java:408)
        at org.glassfish.jersey.model.internal.ComponentBag$5.call(ComponentBag.java:398)
        at org.glassfish.jersey.internal.Errors.process(Errors.java:315)
        at org.glassfish.jersey.internal.Errors.process(Errors.java:297)
        at org.glassfish.jersey.internal.Errors.process(Errors.java:228)
        at org.glassfish.jersey.model.internal.ComponentBag.registerModel(ComponentBag.java:398)
        at org.glassfish.jersey.model.internal.ComponentBag.register(ComponentBag.java:235)
        at org.glassfish.jersey.model.internal.CommonConfig.register(CommonConfig.java:420)
        at org.glassfish.jersey.server.ResourceConfig.register(ResourceConfig.java:425)
        at org.glassfish.jersey.server.ResourceConfig.registerClasses(ResourceConfig.java:501)
        at org.glassfish.jersey.server.ResourceConfig.registerClasses(ResourceConfig.java:528)
        at org.opencb.opencga.catalog.monitor.MonitorService.start(MonitorService.java:107)
        at org.opencb.opencga.app.cli.admin.executors.CatalogCommandExecutor.daemons(CatalogCommandExecutor.java:224)
        at org.opencb.opencga.app.cli.admin.executors.CatalogCommandExecutor.execute(CatalogCommandExecutor.java:86)
        at org.opencb.opencga.app.cli.admin.AdminMain.main(AdminMain.java:129)
Caused by: java.lang.ClassNotFoundException: javax.annotation.Priority
        at jdk.internal.loader.BuiltinClassLoader.loadClass(java.base@9-internal/BuiltinClassLoader.java:366)
        at jdk.internal.loader.ClassLoaders$AppClassLoader.loadClass(java.base@9-internal/ClassLoaders.java:184)
        at java.lang.ClassLoader.loadClass(java.base@9-internal/ClassLoader.java:419)
```

Building against OpenJDK 8 and running on OpenJDK > 10 results in an issue in jjwt as it doesn't support JDK10+: https://github.com/jwtk/jjwt/issues/399
this results in:
```shell
Exception in thread "main" java.lang.NoClassDefFoundError: javax/xml/bind/DatatypeConverter
       at io.jsonwebtoken.impl.Base64Codec.encode(Base64Codec.java:21)
       at io.jsonwebtoken.impl.Base64UrlCodec.encode(Base64UrlCodec.java:22)
       at io.jsonwebtoken.impl.DefaultJwtBuilder.base64UrlEncode(DefaultJwtBuilder.java:350)
       at io.jsonwebtoken.impl.DefaultJwtBuilder.compact(DefaultJwtBuilder.java:296)
       at org.opencb.opencga.catalog.auth.authentication.JwtManager.createJWTToken(JwtManager.java:86)
       at org.opencb.opencga.catalog.auth.authentication.JwtManager.createJWTToken(JwtManager.java:68)
       at org.opencb.opencga.catalog.auth.authentication.AuthenticationManager.createToken(AuthenticationManager.java:105)
       at org.opencb.opencga.catalog.managers.UserManager.login(UserManager.java:430)
       at org.opencb.opencga.app.cli.admin.executors.CatalogCommandExecutor.daemons(CatalogCommandExecutor.java:217)
       at org.opencb.opencga.app.cli.admin.executors.CatalogCommandExecutor.execute(CatalogCommandExecutor.java:86)
       at org.opencb.opencga.app.cli.admin.AdminMain.main(AdminMain.java:129)
Caused by: java.lang.ClassNotFoundException: javax.xml.bind.DatatypeConverter
       at java.base/jdk.internal.loader.BuiltinClassLoader.loadClass(BuiltinClassLoader.java:583)
       at java.base/jdk.internal.loader.ClassLoaders$AppClassLoader.loadClass(ClassLoaders.java:178)
       at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:521)
       ... 11 more
```

**Comment by lawrencegripper** [comment]
We've now sucessfully built the images with Oracle JRE. Need to test if this resolves #1076
